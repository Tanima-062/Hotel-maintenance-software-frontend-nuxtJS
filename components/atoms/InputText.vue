<template>
  <input
    :id="id"
    :type="type"
    class="rounded-md border border-solid text-black-dark placeholder-black-light px-3 focus:shadow-outline focus:outline-none"
    :class="dynamicStyle"
    :placeholder="placeholder"
    :value="value"
    :disabled="disabled"
    @input="input"
    @keydown="isNumber"
    @focus="isFocus = true"
    @blur="blur"
    @keydown.space="(event) => (preventSpace ? event.preventDefault() : event)"
  />
</template>
<script lang="ts">
import { Vue, Component, Prop, Emit } from 'nuxt-property-decorator'

@Component
export default class extends Vue {
  @Prop({ default: null })
  id?: string

  @Prop({ default: 'text' })
  type!: string

  @Prop({ default: '' })
  placeholder!: string

  @Prop({ default: '' })
  value!: string

  @Prop({ default: false })
  error!: boolean

  @Prop({ default: 'bg-white' })
  bg!: string

  @Prop({ default: false })
  disabled!: boolean

  @Prop({ default: 'py-2' })
  padding!: string

  @Prop({ default: false })
  preventSpace!: boolean

  private isFocus: boolean = false

  get dynamicStyle() {
    const classes = []
    // if (this.isFocus) {
    // classes.push('border-blue-600')
    if (this.error) {
      classes.push('border-alert')
      classes.push('bg-error')
    } else {
      classes.push(this.bg)
      classes.push('border-bdColor')
    }
    classes.push(this.padding)
    return classes.join(' ')
  }

  private isNumber(event: Event & { keyCode: number }): boolean {
    if (this.type !== 'number') return true
    const preventKey = [38, 40, 190, 69, 187, 189]
    if (preventKey.includes(event.keyCode)) {
      event.preventDefault()
      return false
    } else if (this.value.length >= Number(this.$attrs.maxlength)) {
      if (event.keyCode !== 8) {
        event.preventDefault()
        return false
      }
      return true
    } else {
      return true
    }
  }

  @Emit()
  private input(e: Event & { target: HTMLInputElement }): string {
    return e.target.value
  }

  @Emit()
  private blur(e: Event & { target: HTMLInputElement }): string {
    this.isFocus = false
    return e.target.value
  }
}
</script>
