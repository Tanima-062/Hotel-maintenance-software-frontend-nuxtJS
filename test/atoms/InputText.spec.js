import { shallowMount } from '@vue/test-utils'
import InputText from '@/components/atoms/InputText.vue'

describe('Atoms InputText Check', () => {
  test('InputText Test', () => {
    const wrapper = shallowMount(InputText)

    // check if it is a vue element
    expect(wrapper.vm).toBeTruthy()

    // check Snapshot
    expect(wrapper.element).toMatchSnapshot()
  })
})
