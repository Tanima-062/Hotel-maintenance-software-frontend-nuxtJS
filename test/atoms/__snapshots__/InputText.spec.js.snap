// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms InputText Check InputText Test 1`] = `
<input
  class="rounded-md border border-solid text-black-dark placeholder-black-light px-3 focus:shadow-outline focus:outline-none bg-white border-bdColor py-2"
  placeholder=""
  type="text"
/>
`;
